state/state.go

```diff
@@ -762,3 +762,18 @@ func (st *state) AllPendingTxs() []*tx.Tx {
 
 	return st.txPool.AllPendingTxs()
 }
+
+func (st *state) IsPruned() bool {
+	return st.store.IsPruned()
```

Comment:
The Public Methods should be guarded by lock.

Reply:
Check: https://github.com/pactus-project/pactus/pull/1420#discussion_r1680422259